### ISSUE TITLE
Allow composition of themes

### ISSUE THREAD

**Issue by tlycken** [dcjones/Gadfly.jl#785]
If this worked, it would be really powerful:

```
theme_base = Theme(line_width=4pt, default_point_size=7pt)
theme_green = Theme(theme_base; default_color=colorant"green")
theme_orange = Theme(theme_base; default_color=colorant"orange")
theme_small_orange = Theme(theme_orange; line_width=1pt)
```

I'm imagining it as the four themes being equivalent to these:

```
theme_base = Theme(line_width=4pt, default_point_size=7pt)
theme_green = Theme(line_width=4pt, default_point_size=7pt, default_color=colorant"green")
theme_orange = Theme(line_width=4pt, default_point_size=7pt, default_color=colorant"orange")
theme_small_orange = Theme(line_width=1pt, default_point_size=7pt, default_color=colorant"orange")
```

**Comment by amellnik** [comment]
Agreed.  I often want to tweak a single setting in a base theme for a specific plot and end up overwriting all the other settings.

**Comment by tlnagy** [comment]
Could you all elaborate a bit more on this suggestion?

**Comment by amellnik** [comment]
Say I have a theme which I use for several graphs with a bunch of options (using the example above):

```julia
theme_base = Theme(line_width=4pt, default_point_size=7pt)
```

Then, if I want to change one setting on a single graph what I would like to do is something like 

```julia
theme_green = Theme(theme_base; default_color=colorant"green")
```

However, this throws an error.  Instead you have to specify every singe option over again and construct a new theme:

```julia
theme_green = Theme(line_width=4pt, default_point_size=7pt, default_color=colorant"green")
```

(This is a moot point for me as I no longer use Gadfly, just expanding on the original issue).

**Comment by tlnagy** [comment]
Ah, I missed the `theme_base;` in the Theme parameters. Thanks for making this clear. Out of curiosity, what do you use now?

**Comment by amellnik** [comment]
I use @spencerlyon2's excellent [PlotlyJS.jl](https://github.com/spencerlyon2/PlotlyJS.jl).  I use the normal Plotly javascript library for my web-based plots and it's very handy to also use the same package in Julia via Jupyter.

**Comment by spencerlyon2** [comment]
Hey @amellnik thanks for the shout out.

Coincidentally I'm actually planning on adding this exact feature to PlotlyJS.jl within the next week!
